src/futures/serial.rs

```diff
@@ -0,0 +1,42 @@
+//! Serial interface
+
+use core::future::Future;
+
+/// Read half of a serial interface
+///
+/// Some serial interfaces support different data sizes (8 bits, 9 bits, etc.);
+/// This can be encoded in this trait via the `Word` type parameter.
+pub trait Read<Word> {
+    /// Read error
+    type Error;
+
+    /// The future associated with the `read` method.
+    type ReadFuture<'a>: Future<Output=Result<Word, Self::Error>> + 'a
+    where
+        Self: 'a;
+
+    /// Reads a single word from the serial interface
+    fn read<'a>(&'a mut self) -> Self::ReadFuture<'a>;
```

Comment:
I still think uart read/write should take slices like all other traits. Writing/reading word-by-word is super slow, especially in nRF chips which can *only* do DMA.

Reply:
How does DMA work when the slices would be different sizes? Is that something DMA on MCUs supports?

Reply:
I think you meant to ask that on SPI?

Reply:
Oops, yep. As for this, I'm fine with that, slice it is.